app/controllers/questions_controller.rb

```diff
@@ -16,15 +16,42 @@ def filter_questions_by_tag(questions, tagnames)
     end
   end
 
+  def period
+    params.slice(:week, :month, :year)
+  end
+
+  def filtering(period)
+    results = self.nil
+    period.each do |key, value|
+      results = results.public_send(key, value) if value.present?
+    end
+    results
+  end
+
   public
 
   def index
+    if params[:period]
```

Comment:
This might be a good one to cache daily! I think esp. with collecting a year's worth of content each page load.

Reply:
great! thanks